### DISCUSSION TITLE
Saving and reloading splink dataframe of predictions

### DISCUSSION THREAD

**Discussion by hunsicke** [moj-analytical-services/splink#1508] [Q&A] [answered]
After predicting with a splink model on a large dataset, is it possible to save the splink dataframe of predictions and reload it later as a splink dataframe for use with .as_record_dict, .waterfall_chart, .comparison_viewer_dashboard, etc?  

What I did so far is convert to pandas then to spark and save as a spark dataframe.  Of course, I can still see the matches, but I can't use the graphical tools for investigating linkages.  So also, is it possible to turn this spark dataframe of predictions back into a splink dataframe?

**Comment by RobinL** [comment, marked as answer]
Yes - if you're using Spark you can do something like:

```python
linker = SparkLinker(df, settings)

df_predict = linker.predict()

spark_df = df_predict.as_spark_dataframe()
spark_df.write.parquet("path")
# later
df_predict_spark = spark.read.parquet("path")
linker.register_table_predict(df_predict_spark)

```

**Reply by hunsicke** [reply]
Yay! it works!!

Thanks!

**Reply by loukak** [reply]
Hi Rob and rest of Splink users,

I have a similar issue in the sense that i want to apply the linker comparison viewer function in not one but three different spark dataframes. All subsets of the first prediction splink dataframe. 

The problem i have is that the comparison viewer function always process the first df that i apply the register_table_predict(). So in your code above the viewer will always process the spark_df even when i register the other two spark dfs.

The comparison viewer is a powerful clerical review tool and all my spark dfs contain different type of links(true matches, below threshold comparisons, conflicts etc) so it would be useful to have a separate dashboard for each one of those. Happy to follow another route if possible.

I use splink v4.0.2

Also what is the difference between register_table() and  register_table_predict() functions?
